example/Router.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+use Equip\Queue\Exception\HandlerException;
+use Equip\Queue\Exception\RouterException;
+use Equip\Queue\RouterInterface;
+
+class Router implements RouterInterface
+{
+    /**
+     * @var array
+     */
+    private $routes;
+
+    /**
+     * @param array $routes
+     */
+    public function __construct(array $routes = [])
+    {
+        $this->routes = $routes;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function get($handler)
+    {
+        $route = $this->getRoute($handler);
+        if (!$route) {
+            throw RouterException::routeNotFound($handler);
+        }
+
+        if (is_callable($route)) {
+            return $route;
+        }
+
+        if (class_exists($route)) {
+            return new $route;
+        }
```

Comment:
Seems like these two branches should look more like this:

```php
if (class_exists($route)) {
    $route = new $route;
}

if (is_callable($route)) {
    return $route;
}
```

That way, if `$handler` is a FQCN, it's instantiated and then the `is_callable()` check is still applied to it.

Not sure if we should allow FQCNs to be passed in or require that they be pre-instantiated objects since DI may be required and I suspect we may not want to couple the DI container to the router implementation.